app/questionnaire/path_finder.py

```diff
@@ -59,19 +59,32 @@ def build_path(self):
                 group_blocks = self._build_blocks_for_group(group, instance_idx)
                 blocks += group_blocks
 
```

Comment:
May be worth commenting this `first_group_instance` related block to say why it's required briefly, e.g. "`first_group_instance` is required in the case that the first instance of a repeating group is skipped"?

Reply:
Changed function name to make it clearer